creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -329,9 +325,19 @@ class SnapbackSM {
         healthyNodes
       })
 
-      this.log(`[issueUpdateReplicaSetOp] Updating userId=${userId} wallet=${wallet} replica set=[${primary},${secondary1},${secondary2}] to new replica set=[${newPrimary},${newSecondary1},${newSecondary2}]`)
+      if (!newPrimary || !newSecondary1 || !newSecondary2) {
+        const errorMsg = `[issueUpdateReplicaSetOp] userId=${userId} wallet=${wallet} Issue with determining new replica set=[${newPrimary},${newSecondary1},${newSecondary2}]`
+        this.logError(errorMsg)
+        response.errorMsg = errorMsg
+        return response
+      }
+
+      this.log(`[issueUpdateReplicaSetOp] Updating userId=${userId} wallet=${wallet} replica set=[${primary},${secondary1},${secondary2}] to new replica set=[${newPrimary},${newSecondary1},${newSecondary2}] issueReconfig=${issueReconfig}`)
 
-      if (!this.snapbackReconfigEnabled) return
+      if (!this.snapbackReconfigEnabled || !issueReconfig) {
```

Comment:
i want to make sure that the configvars give us sufficient control over these ops. seems there are a few diff modes we want to be able to toggle between:
1. never issue reconfig op
2. only issue reconfig op for single unhealthy secondary
3. only issue reconfig ops for secondaries
4. issue reconfig ops for primary or secondary
5. issue reconfig ops in all cases, including entire replica set unhealthy (total data loss)

would be good to think through a simple way to handle this

Reply:
discussed offline - chat tm with team